Mathlib/Algebra/Operad/Basic.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2024 Alex Meiburg. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Meiburg
+-/
+import Mathlib.Algebra.Group.Action.Defs
+
+/-! This file defines the typeclasses and notations used for workign with `Operad`, `SymmOperad`,
+  and `Clone`. These all, in some form another, work with graded - or Sigma - types. Below,
+  `A : ℕ → Type*` is always the carrier.
+
+  Classes:
+  - `MultiComposable`: A `compose` operation `A n → Fin n → A m → A (n+m-1)`, for composing an
+    element of grade `m` into one of grade `n` at a location `Fin n`. This is the type of operation
+    used in Operads.
+  - `Superposable`: A `superpose` operation `A n → (Fin n → A m) → A m`. This is the type of
+    operation used in Clones.
+  - `OneGradedOne`: There is a distinguished `1` element at grade 1, a notion of identity.
+  - `SigmaMulAction`: A graded `MulAction M A`, where elements `M i` can act on `A i` at each i.
+
+  Definitions:
+  - `composeAt` bundles the action of `MultiComposable.compose` into `Sigma A`.
+  - `superpose` bundles the action of `Superposable.superpose` into `Sigma A`.
+
+  Notations:
+  - `x ∘⟨p⟩ y` for `MultiComposable.compose` to compose `x ∘ y` at index `p`.
+  - `x ∘[p] y` for the `composeAt` Sigma variant.
+  - `x ∘⚟ y` for `Superposable.superpose x y`. The typography is meant to suggest
+    "many arguments into one".
+-/
```

Comment:
Could you add references to the mathematical literature so that the reviewers (who are not very much expert in this field) may compare the definitions?

Reply:
Done